### PR TITLE
#3307  SensorStatus

### LINKED ISSUE #3307: SensorStatus component
*EPIC: #3161*

*Related Marvel Designs: https://marvelapp.com/prototype/55aea8f*
*Related Component Breakdown: https://app.diagrams.net/#G1woR4jOSMkfeHP_efsGrW5lU35pVmKvCR*

## Is your feature request related to a problem? Please describe.

Don't have the component for `SensorStatus` as shown in the component breakdown

## Describe the solution you'd like

Take/adapt the component from CCE: https://github.com/openmsupply/msupply-cold-chain/blob/develop/src/ui/components/SensorStatus.js

## Implementation

N/A

## Describe alternatives you've considered

N/A

## Additional context

N/A

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #3307 

## Change summary

- Adds `SensorStatus` component

## Testing

Internal

### Related areas to think about

sensorstatus

### CODE REVIEW

File: src/utilities/constants.js [hunk 1]
```diff
@@ -7,5 +7,5 @@ export const MILLISECONDS = {
   THIRTY_MINUTES: 1000 * 60 * 30,
   FIVE_MINUTES: 1000 * 60 * 5,
   ONE_DAY: 1000 * 60 * 60 * 24,
-  ONE: 1000,
+  ONE_SECOND: 1000,
```
Comment (joshxg) [inline src/utilities/constants.js:10]: Renamed this and this needed the change in the only place it was used `TemperatureSyncActions`

File: src/widgets/Rectangle.js [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { DANGER_RED } from '../globalStyles';
+
+const SIZES = {
+  large: { width: 180, height: 60 },
+};
+
+export const Rectangle = ({ children, colour, size }) => {
```
Comment (joshxg) [inline src/widgets/Rectangle.js:11]: This will be utilized in at least one other place

File: src/widgets/Rectangle.js [hunk 3]
```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+
+import { DANGER_RED } from '../globalStyles';
+
+const SIZES = {
+  large: { width: 180, height: 60 },
```
Comment (joshxg) [inline src/widgets/Rectangle.js:8]: didn't want to make different sizes but makes this more open to change

File: src/widgets/SensorStatus.js [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+    // Start an infinite looping sequence of animations.
+    animRef.current = Animated.loop(
+      // Sequence is a a sequence of sequences - each sub sequence being
+      // for a specific icon which should be shown. i.e. if the fridge is
+      // low battery and has a hot breach, there is a sequence of sequences
+      // for each icon.
+      Animated.sequence(
+        // Map each animation value (which can be thought of as an icon)
+        // to a sequence which will increase the opacity of the icon to full
+        // over one second, then reduce the opacity to zero over one second.
```
Comment (joshxg) [inline src/widgets/SensorStatus.js:47]: I don't really like so much commenting I think it makes the code harder to read. Please let me know if it is too much as a reader or if it was helpful.

**Review by mark-prins — APPROVED** [review]
Very nice!

**Pull request merged**